### PR TITLE
feat: add support to upload file via file explorer

### LINKED ISSUE #30: Add support for uploading file through file explorer

### PR DESCRIPTION
**Pull request by jienius**
[Please delete all explanatory text in square brackets prior to creating the pull request!]
[Write a short explanation of your changes]

[Link to an existing issue number]
Fixes #30  
 OR
Resolves #

### CODE REVIEW

File: src/app/Upload.svelte [hunk 1]
```diff
@@ -30,55 +29,74 @@
...
-  on:dragover={handleDragOver}
-  on:dragleave={handleDragLeave}>
-  <div class='inner-border' class:drag-over={onDraggedOver}>
-    <p>Drag file to upload ...</p>
-  </div>
+  class:drag-over={onDraggedOver}
+  on:drop|preventDefault={handleDrop}
+  on:dragover|preventDefault={handleDragOver}
+  on:dragleave|preventDefault={handleDragLeave}>
+  <img src="icons/upload-solid.svg" alt="upload icon" />
```
Comment (eddieantonio) [inline src/app/Upload.svelte]: Mark this as a decorative image:

```suggestion
  <img role="presentation src="icons/upload-solid.svg" alt=""  />
```

See to: https://www.w3.org/WAI/tutorials/images/decorative/#decorative-image-as-part-of-a-text-link

File: src/app/Upload.svelte [hunk 2]
```diff
@@ -30,55 +29,74 @@
...
-  on:drop={handleDrop}
-  on:dragover={handleDragOver}
-  on:dragleave={handleDragLeave}>
-  <div class='inner-border' class:drag-over={onDraggedOver}>
-    <p>Drag file to upload ...</p>
-  </div>
+  class:drag-over={onDraggedOver}
+  on:drop|preventDefault={handleDrop}
+  on:dragover|preventDefault={handleDragOver}
+  on:dragleave|preventDefault={handleDragLeave}>
```
Comment (eddieantonio) [inline src/app/Upload.svelte:101]: This is nice!

File: src/app/Upload.svelte [hunk 3]
```diff
@@ -30,55 +29,74 @@
...
-    // Prevent default behavior (Prevent file from being opened)
-    event.preventDefault();
+  };
+
+  const handleChange = (event: Event) => {
+    const input = event.target as HTMLInputElement;
+    if (input !== null && input.files) {
+      for (let file of input.files) {
+        files.set(file.name, file);
+        alert(file.name);
```
Comment (eddieantonio) [inline src/app/Upload.svelte:45]: So eventually, we will _not_ use `alert()`, but this is good for now!

File: src/app/Upload.svelte [hunk 4]
```diff
@@ -1,13 +1,12 @@
 <script lang="ts">
+  const UPLOAD_INPUT_ID = "upload-input";
```
Comment (eddieantonio) [inline src/app/Upload.svelte:2]: I like this!

**Review by eddieantonio — APPROVED** [review]
LGTM! There's just one change you can make for accessibility: essentially telling assistive technologies that the upload icon is purely decorative.

File: src/app/App.svelte [hunk 5]
```diff
@@ -5,6 +5,23 @@
   let files: any;
 </script>
 
+<style>
+  :global(:root) {
+    --blue: #6fbbd4;
+    --blue-light: #b3dbe7;
```
Comment (jienius) [inline src/app/App.svelte:11]: Copied from that PPT

**Pull request merged**